--- scripts/git-hooks/pre-commit	2021-03-10 13:30:08.000000000 -0500
+++ scripts/git-hooks/pre-commit	2021-10-21 15:09:00.446062448 -0400
@@ -7,5 +7,12 @@
 git diff-index --quiet HEAD -- || echo "Untracked files are present, this may impact the validity of pre-commit checks."
 "${ROOT_DIR}"/vendor/bin/blt internal:git-hook:execute:pre-commit "${LIST}"
 
+if [[ -f  "${ROOT_DIR}"/vendor/bin/phpcs ]]; then
+  for i in ${LIST}
+  do 
+    "${ROOT_DIR}"/vendor/bin/phpcs $i
+  done 
+fi
+
 # Return the status of the last run command.
 exit $?
